swarm/src/lib.rs

```diff
@@ -1580,8 +1580,10 @@ impl fmt::Display for DialError {
                 f,
                 "Dial error: tried to dial local peer id at {endpoint:?}."
             ),
-            DialError::DialPeerConditionFalse(c) => {
-                write!(f, "Dial error: condition {c:?} for dialing peer was false.")
+            DialError::DialPeerConditionFalse(condition) => match condition {
```

Comment:
This inner match can be embedded in the outer one:

```suggestion
            DialError::DialPeerConditionFalse(PeerCondition::Disconnected) => {
```

That will reduce indentation of this block :)